src/main/java/application/DebuggerController.java

```diff
@@ -0,0 +1,52 @@
+package application;
+
+import auth.DjangoAuth;
+import beans.SessionRequestBean;
+import beans.debugger.DebuggerFormattedQuestionsResponseBean;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import objects.SerializableFormSession;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.web.bind.annotation.*;
+import repo.SerializableMenuSession;
+import services.FormattedQuestionsService;
+import session.FormSession;
+import session.MenuSession;
+import util.Constants;
+
+/**
+ * Controller class for all routes pertaining to the CloudCare Debugger
+ */
+@Api(value = "Debugger Controller", description = "Operations involving the CloudCare Debugger")
+@RestController
+@EnableAutoConfiguration
+public class DebuggerController extends AbstractBaseController {
+
+    @Autowired
+    private FormattedQuestionsService formattedQuestionsService;
+
+    @ApiOperation(value = "Get formatted questions and instance xml")
+    @RequestMapping(value = Constants.URL_DEBUGGER_FORMATTED_QUESTIONS, method = RequestMethod.POST)
+    public DebuggerFormattedQuestionsResponseBean getFormattedQuesitons(
+            @RequestBody SessionRequestBean debuggerRequest,
+            @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+
+        SerializableFormSession serializableFormSession = formSessionRepo.findOneWrapped(debuggerRequest.getSessionId());
+        FormSession formSession = new FormSession(serializableFormSession);
+        SerializableMenuSession serializableMenuSession = menuSessionRepo.findOne(serializableFormSession.getMenuSessionId());
```

Comment:
gotcha that makes sense. ok agree, the app id in the form session would be hugely helpful